src/ng/filter/filter.js

```diff
@@ -127,7 +127,7 @@
  */
 function filterFilter() {
   return function(array, expression, comparator) {
-    if (!isArray(array)) {
+    if (!isArrayLike(array)) {
```

Comment:
I've updated the jsperf (http://jsperf.com/with-or-without-isarray/2) to reuse `Array.prototype.filter` (although that should hardly make any noticable difference).

Basically, 99% of the time, I believe we will be dealing with arrays, so that is the usecase we should be optimizing for. So, we are mainly concerned about `array with check` vs `array without check` (using `Array.prototype.filter`).
Indeed, on desktop Chrome and Firefox, both seem to be almost equally fast.
On mobile Chrome, the latter was about 10% slower.
More importantly, on desktop IE11, the latter was consistently >15% slower.

So, I would go with the former.

---
We should: 

1.) Store `Array.prototype.filter` (if nothing else for readability)
2.) Allow `array-like` objects.
3.) Check `array` and decide what function to call.

```js
var arrayFilter = Array.prototype.filter;
...
if (!isArrayLike(array)) { /* Throw */ }
...
return isArray(array) ? array.filter(predicateFn) : arrayFilter.call(array, predicateFn);
```